### PR TITLE
refactor(state-transition): re-organize test helpers

### PR DESCRIPTION
**Pull request by spiral-ladder**
**Motivation**

#10090 lints and errors for packages importing something from a dev dependency.

testFileCache.ts imports `getClient` from `@lodestar/api`, which is only a devDependency of state-transition, yet src/testUtils/index.ts re-exported it through the published ./test-utils entry. Any consumer of that entry would need @lodestar/api installed. Its only users are three perf tests in this package, so move it to test/utils and drop the re-export instead of adding the API client as a runtime dependency of a core library.

**Description**

This PR reorganizes test helpers by getting rid of re-exporting `testFileCache.ts` by separating concerns with the new `networkCachedState.ts` and `networkCachedStateAndBlock.ts` test helpers in their respective packages which consume `testFileCache.ts` which only owns the download/cache logic

unblocks #10090

**AI Assistance Disclosure**

fixed w claude

### CODE REVIEW

File: packages/test-utils/src/testFileCache.ts [hunk 1]
```diff
@@ -99,7 +99,6 @@ export async function getNetworkCachedBlock(
 
 async function downloadTestFile(fileId: string): Promise<Uint8Array> {
   const fileUrl = `${TEST_FILES_BASE_URL}/${fileId}`;
-  // biome-ignore lint/suspicious/noConsole: We explicity need to log to console
```
Comment (spiral-ladder) [inline packages/test-utils/src/testFileCache.ts:102]: this is no longer needed now that this is no longer in src

**Comment by spiral-ladder** [comment]
edited description + title bc this turned into reorganization instead across diff packages

**Comment by nflaig** [comment]
@lodekeeper please review

**Review by lodekeeper — APPROVED** [review]
Reviewed current head `995a051f` after the follow-up commits. The two earlier concerns are addressed: `packages/state-transition/src/testUtils/index.ts` no longer re-exports the cache helper, and the remaining beacon-node perf consumer now imports its local wrapper from `test/utils/networkCachedStateAndBlock.js`, so the removed `@lodestar/state-transition/test-utils` export is no longer needed there. The shared byte-level cache/download logic is in `@lodestar/test-utils`, with package deps updated accordingly.

CI is green on the current head. Looks good to me.

File: packages/test-utils/src/testFileCache.ts [hunk 2]
```diff
@@ -115,6 +113,15 @@ async function downloadTestFile(fileId: string): Promise<Uint8Array> {
...
 }
 
+function getInfuraBeaconUrl(network: NetworkName): string {
+  const credentials = process.env.INFURA_ETH2_CREDENTIALS;
+  if (!credentials) {
+    throw Error("Must set ENV INFURA_ETH2_CREDENTIALS");
+  }
+
+  return `https://${credentials}@eth2-beacon-${network}.infura.io`;
+}
```
Comment (nflaig) [inline packages/test-utils/src/testFileCache.ts:123]: this code is 100% unused?

File: packages/test-utils/src/testFileCache.ts [hunk 3]
```diff
@@ -115,6 +113,15 @@ async function downloadTestFile(fileId: string): Promise<Uint8Array> {
   }
 }
 
+function getInfuraBeaconUrl(network: NetworkName): string {
```
Comment (nazarhussain) [inline packages/test-utils/src/testFileCache.ts:116]: `getInfuraBeaconUrl` this is duplicate now in `packages/test-utils/src/testFileCache.ts` and `packages/state-transition/src/testUtils/infura.ts`

File: packages/beacon-node/test/utils/networkCachedStateAndBlock.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import {pubkeyCache} from "@chainsafe/lodestar-z/pubkeys";
+import {NetworkName} from "@lodestar/config/networks";
+import {CachedBeaconStateAllForks} from "@lodestar/state-transition";
+import {testCachePath} from "@lodestar/state-transition/test-utils";
+import {getNetworkCachedBlockBytes, getNetworkCachedStateBytes} from "@lodestar/test-utils";
+import {SignedBeaconBlock} from "@lodestar/types";
+import {createCachedBeaconStateTest} from "./cachedBeaconState.js";
+
+export async function getNetworkCachedState(
```
Comment (nazarhussain) [inline packages/beacon-node/test/utils/networkCachedStateAndBlock.ts:9]: This function is also duplicate. Currently they differ in the way `createCachedBeaconStateTest` is called internally, but the both can drift easily in future. Would be nice if we can also remove that duplicate.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: packages/test-utils/src/testFileCache.ts (hunk 3 above)
Comment (nflaig) [reply packages/test-utils/src/testFileCache.ts:116]: can we please delete this function, I hope no test we run currently in ci relies on this, otherwise, yeet that one too

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts:116]: I think this function is used in `analyzeBlocks` and `analyzeEpochs` (which aren't automatically ran as benchmarks but as manual scripts), do we still use those?

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts:116]: @lodekeeper can you check for the last time both those functions were used for analysis in our github history or discord chat logs?

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts:116]: Seems like the last time we had any mention of them on github was in 2021 in the PRs that introduced them, if these scripts are no longer useful, we can remove them and therefore this function. cc @wemeetagain since he was tagged in reviews for those

https://github.com/ChainSafe/lodestar/pull/2934
https://github.com/ChainSafe/lodestar/pull/2979

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts:116]: Verified the call graph on `995a051f` — and there's one wrinkle worth flagging before anything is deleted: the two copies have **different consumers**, so they're not interchangeable.

- **This copy** (`test-utils/src/testFileCache.ts`) is *live*. It's the private helper behind `getNetworkCachedStateBytes` / `getNetworkCachedBlockBytes` → `getNetworkCachedState`, which the perf suite runs: `epochAltair` / `epochCapella` / `epochPhase0.test.ts` and `verifyImportBlocks.test.ts`. So per @nflaig's "I hope no test we run in ci relies on this, otherwise yeet that one too" — CI *does* rely on this one, so it stays.
- **The other copy** (`state-transition/src/testUtils/infura.ts`) is the one used only by `analyzeBlocks` / `analyzeEpochs`. Confirmed dead: last real touch is the 2021 PRs you found (#2934 / #2979), and nothing on the branch runs them — no vitest/package.json/CI wiring, only a stale doc-comment pointer in `processAttestationPhase0.ts`.

So the dedup that matches the directive = keep the test-utils copy, yeet the state-transition side:
- `packages/state-transition/src/testUtils/infura.ts` (holds only `getInfuraBeaconUrl`, exported nowhere else)
- `packages/state-transition/test/perf/analyzeBlocks.ts`
- `packages/state-transition/test/perf/analyzeEpochs.ts`
- `packages/state-transition/test/perf/csv.ts` (imported only by `analyzeEpochs` → orphaned once it's gone)

That leaves `getInfuraBeaconUrl` single-sourced in `test-utils`. I don't have write access to your branch to push it, so it's yours to drop in — or happy to open a follow-up PR against `unstable` if you'd rather I take it.

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts:116]: Opened https://github.com/ChainSafe/lodestar/pull/10124 that yeets the state transition side

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts:116]: > It's the private helper behind getNetworkCachedStateBytes / getNetworkCachedBlockBytes → getNetworkCachedState, which the perf suite runs: epochAltair / epochCapella / epochPhase0.test.ts and verifyImportBlocks.test.ts

Now we should decide if we want to keep these, are we still using those perf benchs to make any decisions? if we are, we might want to come up with an alternative instead of simply yeeting

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts:116]: Thanks for picking it up. Reviewed #10124 — it's exactly the call-graph result: drops the `state-transition` `getInfuraBeaconUrl` copy plus the three dead scripts (`analyzeBlocks` / `analyzeEpochs` / `csv`), and even cleans the now-stale `analyzeBlocks.ts` pointer in the `processAttestationPhase0` doc-comment. `getInfuraBeaconUrl` ends up single-sourced in `test-utils`. LGTM from me (left an approving review there); +@wemeetagain for the final sign-off since he approved the original #2934/#2979.

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts:116]: Good question — but these are in a different bucket from the analyze scripts. The `epoch{Phase0,Altair,Capella}` and `verifyImportBlocks` perf tests aren't manual: they're picked up automatically by `pnpm benchmark` (glob `packages/*/test/perf/**/*.test.ts`) and run in CI via `benchmark.yml` on every PR — with `INFURA_ETH2_CREDENTIALS` + a cached perf-state, and `BENCHMARK_S3=true` writing a rolling baseline vs `unstable` and posting the comparison back onto the PR. So they're live CI, not dead code — which is exactly why keeping the `test-utils` copy of `getInfuraBeaconUrl` is the right call, and why #10124 correctly leaves them alone.

Whether we still *act* on those epoch-transition numbers (vs skimming past them) is a fair thing to reassess — but that's a separate maintainer call from this dedup, and I wouldn't fold a "replace the perf benches" change into #10124/#10099. If it's worth revisiting, it probably deserves its own issue. @nflaig / @wemeetagain — do you still treat the epoch-transition benches as real signal, or is that worth a rethink?

Comment (nflaig) [reply packages/test-utils/src/testFileCache.ts:116]: > I think this function is used in `analyzeBlocks` and `analyzeEpochs` (which aren't automatically ran as benchmarks but as manual scripts), do we still use those?

must be something that was added pre-merge, so stale for 4 years at least

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts:116]: Yep — dead since the 2021 PRs that added them. Already being removed in #10124 (the `state-transition` `getInfuraBeaconUrl` copy + the 3 dependent scripts); just waiting on the original reviewers to sign off there, per your note on that PR.

**Review by nazarhussain — DISMISSED** [review]

File: packages/test-utils/src/testFileCache.ts (hunk 2 above)
Comment (nflaig) [inline packages/test-utils/src/testFileCache.ts]: why is this still here?

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: It's dead — safe to remove. `getInfuraBeaconUrl` builds `https://…@eth2-beacon-<network>.infura.io`, but Infura sunset those endpoints: `eth2-beacon-{mainnet,holesky,sepolia}.infura.io` all NXDOMAIN now. It's only the 2nd `tryEach` fallback in `getNetworkCachedStateBytes` / `getNetworkCachedBlockBytes`, reached only if the primary `downloadTestFile` (dapplion GH-release fetch) fails — and if it ever were reached, `getClient` would just throw on DNS. So the perf tests already rely entirely on the GH-release download + local cache; the Infura branch can't succeed.

So it can all go (all in `test-utils/src/testFileCache.ts`): `getInfuraBeaconUrl` + the Infura branch of both `tryEach` calls (each then collapses to a single `downloadTestFile`, so `tryEach` goes too), the now-unused `getClient` import (`@lodestar/api`) and the `timeout?` params (they only fed the Infura client), plus `INFURA_ETH2_CREDENTIALS` in `benchmark.yml`.

This file only exists on this branch (the reorg introduces it), so easiest is to fold the cleanup into this PR — @spiral-ladder want to take it, or should I open a follow-up once this merges? (No push access to your branch here.)

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts]: we still rely on `getNetworkCachedState` (and its bytes equivalent) for perf benchmarks (see [epochCapella.ts](https://github.com/ChainSafe/lodestar/blob/92dfaf8f5323e823ce8249eaaa9db3966942b33e/packages/state-transition/test/perf/epoch/epochCapella.test.ts#L24))

Dependency goes like this: epochCapella -> getNetworkCachedState -> getNetworkCachedStateBytes (in `@lodestar/test-utils`, this file) -> getInfuraBeaconUrl

[was asking](https://github.com/ChainSafe/lodestar/pull/10099#discussion_r4043785210) if we were still relying on these benchmarks to make any decisions, otherwise we can just delete those too

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Yep, you are right on the live dependency: the epoch/verify-import perf benches still go through `getNetworkCachedState(Bytes)` / `getNetworkCachedBlock(Bytes)`, so I would not delete those benchmarks as part of this cleanup.

The distinction I was trying to draw is narrower: the live benchmark path should stay, but the `getInfuraBeaconUrl` fallback underneath it is dead. `downloadTestFile()` is the path that supplies the cached artifacts; the Infura branch is only the second fallback inside `tryEach`, and those `eth2-beacon-<network>.infura.io` endpoints are gone/NXDOMAIN now. So removing the Infura fallback does not require deleting the benchmarks.

I checked out this PR head and made the exact cleanup locally, but pushing to `ChainSafe/lodestar:bing/move-test-file-cache` is denied for `lodekeeper` (403). The patch I would fold into this PR is:

- in `packages/test-utils/src/testFileCache.ts`: remove `getClient`, `getInfuraBeaconUrl`, `tryEach`, and both Infura fallback branches; have `getNetworkCachedStateBytes` / `getNetworkCachedBlockBytes` call `downloadTestFile(fileId)` directly; drop their `timeout?` params
- in `packages/state-transition/test/utils/networkCachedState.ts` and `packages/beacon-node/test/utils/networkCachedStateAndBlock.ts`: drop the now-unused timeout plumbing into the bytes helpers
- in the perf callers: remove the third argument from `getNetworkCachedState(...)` / `getNetworkCachedBlock(...)`, while keeping the outer `beforeValue(..., 300_000)` setup timeout
- in `.github/workflows/benchmark.yml`: remove `INFURA_ETH2_CREDENTIALS`

Local verification for that patch: `pnpm lint`, `pnpm build`, and `pnpm check-types` all pass.

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts]: honestly, looking at it again, i think i will just delete those, the capella one seems wrong anyway because [we're using `ForkSeq.altair`](https://github.com/ChainSafe/lodestar/blob/57ffc39074bb3888e43ec436a72460f94aa34343/packages/state-transition/test/perf/epoch/epochCapella.test.ts#L29) for some reason, so it seems like it doesn't matter anyway

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Good catch on `epochCapella` — that one is stale/wrong, but I would fix it rather than use it as the reason to delete the whole benchmark path.

I re-checked the current PR head (`92dfaf8f`) against `processEpoch`. With `ForkSeq.altair`, the top-level Capella benchmark skips the Capella branch:

- `processEpoch` uses `fork >= ForkSeq.capella` to choose `processHistoricalSummariesUpdate(...)`; otherwise it runs `processHistoricalRootsUpdate(...)`.
- So the current Capella perf bench is running a Capella state through the Altair historical-roots branch. The per-step section has the same stale bit: it benchmarks `processHistoricalRootsUpdate`, while the Capella path should be `processHistoricalSummariesUpdate`.

But that points to a narrow benchmark fix, not deleting the live benchmarks as part of the test-file-cache cleanup. The exact patch I would apply is:

- in `packages/state-transition/test/perf/epoch/epochCapella.test.ts`, change `const fork = ForkSeq.altair` to `ForkSeq.capella`
- replace the `processHistoricalRootsUpdate` import/step with `processHistoricalSummariesUpdate`

I applied that locally on top of the dead-Infura cleanup as signed commit `e8b4aeb032` (previous cleanup commit was `d1ceacf76b`), but I still cannot push to `ChainSafe/lodestar:bing/move-test-file-cache` — same 403 for `lodekeeper`.

Verification for the combined local patch now passes: `pnpm lint`, `pnpm check-types`, and `pnpm build`.

So my recommendation: fold in the dead Infura fallback removal plus this one-line-ish Capella benchmark fix. If maintainers want to retire the epoch/verify-import perf benches entirely, I would do that as a separate benchmark policy cleanup, not as a side effect of this helper reorg.

Comment (spiral-ladder) [reply packages/test-utils/src/testFileCache.ts]: @lodekeeper re-review the dead infura fallback removal

also open a PR fixing the capella benchmark

**Comment by spiral-ladder** [comment]
i think we also want to remove our reliance on `TEST_FILES_BASE_URL`, are there any reasons why we need to fetch blocks for perf tests and `verifyImportBlock.test.ts`? If those are not relied upon anymore we can yeet those perf tests too, I saw `verifyImportBlock.test.ts` had recent edits, im assuming that is still used

File: packages/test-utils/src/testFileCache.ts (hunk 2 above)
Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Re-reviewed the current #10099 head (`bbe68e5d`) for the dead Infura fallback removal.

The actual fallback removal looks right to me now: `getInfuraBeaconUrl`, the `@lodestar/api` client fallback, `tryEach`, timeout plumbing, and the `INFURA_ETH2_CREDENTIALS` benchmark secret are gone. The remaining fetch path is the existing GitHub release fixture source, moved into `@lodestar/test-utils` as byte-level helpers.

One small cleanup I would still fold into #10099: the new `getNetworkCachedStateBytes` / `getNetworkCachedBlockBytes` comments still say "Download a state from Infura." Those should say they download/cache the test fixture bytes, since Infura is no longer involved.

I also opened the separate Capella benchmark fix PR here: https://github.com/ChainSafe/lodestar/pull/10145

**Comment by lodekeeper** [comment]
I re-checked the current uses.

`TEST_FILES_BASE_URL` is still the source for the deterministic SSZ fixture files used by the perf helpers after the dead Infura fallback is removed. In #10099, the dependency becomes:

- epoch perf benches -> `getNetworkCachedStateBytes(...)` -> GitHub release fixture URL
- `verifyImportBlocks.test.ts` -> `getNetworkCachedStateBytes(...)` / `getNetworkCachedBlockBytes(...)` -> GitHub release fixture URL

So I would separate these decisions:

1. Remove Infura fallback: yes, #10099 does that and I think the code shape is right, modulo stale comments that still mention Infura.
2. Remove `TEST_FILES_BASE_URL`: that means either deleting these perf fixtures or replacing the fixture source/mirroring strategy. I would not do that as part of the dead-Infura cleanup unless we explicitly decide to retire the epoch/range-sync perf benches.
3. `verifyImportBlocks.test.ts`: it is skipped/manual, but it is still maintained and was recently updated, so I would not treat it as dead just because it fetches block fixtures.

For the stale Capella benchmark, I opened the separate fix PR requested here: https://github.com/ChainSafe/lodestar/pull/10145

File: packages/test-utils/src/testFileCache.ts (hunk 3 above)
Comment (wemeetagain) [reply packages/test-utils/src/testFileCache.ts]: yes happy to remove it

Comment (wemeetagain) [reply packages/test-utils/src/testFileCache.ts]: imo its desirable to remove the infura provider entirely

Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Agreed. That changes the direction from the earlier "keep the live `test-utils` fallback" split: remove the Infura provider entirely, but keep the deterministic `TEST_FILES_BASE_URL` fixture download path for the perf/verify-import consumers.

I already have this as local commit `d1ceacf76b` on top of this PR head from the earlier pass: it removes `getInfuraBeaconUrl`, the `@lodestar/api` client fallback, `tryEach`, Infura timeout plumbing, and the `INFURA_ETH2_CREDENTIALS` benchmark secret. I still cannot push directly to this branch (`maintainerCanModify=false` / 403), so the actionable patch is ready for the PR author/maintainers to fold in.

**Review by wemeetagain — APPROVED** [review]
LGTM

File: packages/test-utils/src/testFileCache.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import {ChainForkConfig, createChainForkConfig} from "@lodestar/config";
+import {NetworkName, networksChainConfig} from "@lodestar/config/networks";
+import {fetch} from "@lodestar/utils";
+
+/**
+ * Full link example:
+ * ```
+ * https://github.com/dapplion/ethereum-consensus-test-data/releases/download/v0.1.0/block_mainnet_3766821.ssz
+ * ``` */
+const TEST_FILES_BASE_URL = "https://github.com/dapplion/ethereum-consensus-test-data/releases/download/v0.1.0";
```
Comment (nflaig) [inline packages/test-utils/src/testFileCache.ts]: bit strange url

File: packages/test-utils/src/testFileCache.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  const stateSsz = await downloadTestFile(fileId);
+
+  fs.writeFileSync(filepath, stateSsz);
+  return {config, bytes: stateSsz};
+}
+
+/**
+ * Download a state from Infura. Caches states in local fs by network and slot to only download once.
+ */
+export async function getNetworkCachedBlockBytes(
```
Comment (nflaig) [inline packages/test-utils/src/testFileCache.ts]: @lodekeeper what are doing with this? is it used?

File: packages/test-utils/src/testFileCache.ts (hunk 5 above)
Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Fair — it's a personal-repo release (`dapplion/ethereum-consensus-test-data` v0.1.0), not a ChainSafe-owned location. It isn't new in this PR though: the constant is moved verbatim from `packages/state-transition/src/testUtils/testFileCache.ts:19` on unstable (same URL, same `${TEST_FILES_BASE_URL}/${fileId}` fetch), and it still resolves (HTTP 200 for `block_mainnet_3766821.ssz`). Since this PR is a pure reorg I'd leave the URL as-is here.

If you want it off a personal account, the follow-up is mirroring the `state_<network>_<slot>.ssz` / `block_<network>_<slot>.ssz` fixtures to a ChainSafe-owned release/bucket and changing this one constant — tell me where they should live and I'll prep it.

File: packages/test-utils/src/testFileCache.ts (hunk 6 above)
Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Only by a skipped test. `getNetworkCachedBlockBytes` has one caller, `getNetworkCachedBlock` (`packages/beacon-node/test/utils/networkCachedStateAndBlock.ts:15`), and that has one consumer, `packages/beacon-node/test/perf/chain/verifyImportBlocks.test.ts:60` — inside `describe.skip("verify+import blocks - range sync perf test")` (line 29; the skip is already on unstable). So nothing executes the block path, it only has to keep compiling. `getNetworkCachedStateBytes` just above it is live (phase0/altair/capella epoch perf benches).

Correction to my earlier comment on this PR: I listed `verifyImportBlocks.test.ts` as a live consumer of the fixture download — it's `describe.skip`, so that dependency is dead weight.

Options: (a) drop `getNetworkCachedBlockBytes` + `getNetworkCachedBlock` together with the skipped `verifyImportBlocks.test.ts` perf suite, leaving only the state fixtures on `TEST_FILES_BASE_URL`; or (b) keep it as-is in this reorg. I'd lean (a) as a separate small PR rather than widening this one — say the word and I'll open it.

Comment (nflaig) [reply packages/test-utils/src/testFileCache.ts]: @lodekeeper does this look good can we merge it or should we clean up stuff before, if so please open a pr against this one

File: packages/test-utils/src/testFileCache.ts (hunk 5 above)
Comment (nflaig) [reply packages/test-utils/src/testFileCache.ts]: given that the url was just moved from `packages/state-transition/src/testUtils/testFileCache.ts`, let's reconsider cleaning this up later

**Review by nflaig — APPROVED** [review]

File: packages/test-utils/src/testFileCache.ts (hunk 6 above)
Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Late reply — the run that opened #10174 died before it could post this.

Verdict at `dbdb4ae912`: correct and mergeable as-is. Three leftovers were cheaper to drop before merge than after: (1) the dead block-fixture helpers (`getNetworkCachedBlockBytes` → `getNetworkCachedBlock` → the `describe.skip` `verifyImportBlocks.test.ts` suite, which also drops the new beacon-node `@lodestar/test-utils` devDep and the duplicate `getNetworkCachedState`), (2) the stale "Download a state from Infura" JSDoc, (3) the orphaned `@lodestar/api` devDep in `state-transition`. I opened that as a stacked deletion-only PR, https://github.com/ChainSafe/lodestar/pull/10174 (6 files, +1/-202), but #10099 merged first (`c535e94f25`) and deleting `bing/move-test-file-cache` auto-closed it — nothing wrong with the PR itself.

`lodekeeper:chore/drop-dead-test-fixture-helpers` is intact and merges cleanly onto current `unstable`. If you want the cleanup, say the word and I'll re-open it against `unstable`; otherwise I'll leave it.

File: packages/test-utils/src/testFileCache.ts (hunk 5 above)
Comment (lodekeeper) [reply packages/test-utils/src/testFileCache.ts]: Ack — `TEST_FILES_BASE_URL` stays as-is (moved verbatim, still resolves). Whenever you want to revisit, the follow-up is mirroring the fixtures to a ChainSafe-owned release and changing this one constant.
